build.cmd

```diff
@@ -6,8 +6,26 @@ setlocal
 ::       means that that rebuilding cannot successfully delete the task
 ::       assembly. 
 
-echo %2 | find /I "outerloop"
-if "%errorlevel%" equ "0" (
+set outloop=false
+
+:START_CMDLINE_PARSE
+set SWITCH=%1
+if {%SWITCH%} == {} goto :END_CMDLINE_PARSE
+set VALUE=%2
+
+if /i "%SWITCH%" == "/p:WithCategories" (
+  if /i "%VALUE%" == "OuterLoop" (
+      set outloop=true
+      goto :END_CMDLINE_PARSE
+  )
+)
+
+SHIFT
+SHIFT
```

Comment:
I believe msbuild only does parameters where you do /p:ParameterName=Value - no spaces unless it's quoted. 

For example, 

* `/p:WithCategories=OuterLoop` will make parameter `WithCategories` equal "OuterLoop"
* `/p:WithCategories=Outer Loop` will make parameter `WithCategories` equal to "Outer"
* `/p:WithCategores="outer loop"` will make parameter `WithCategories` equal to "outer loop" 

Reminder that "OuterLoop" is case *insensitive*